src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +461,53 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Entering();
+
+            if (!command.GetEnableAutoLogon())
+            {
+                return;
+            }
+
+            Trace.Info("Agent is going to run as process setting up the 'AutoLogon' capability for the agent.");
+            try
+            {                
+                var autoLogonConfigManager = HostContext.GetService<IAutoLogonConfigurationManager>();
+                autoLogonConfigManager.Configure(command);
+
+                if (autoLogonConfigManager.RestartNeeded())
```

Comment:
RestartNeeded() should be deleted. We should always prompt for restart at the end of auto-logon configure.